src/classes/AutoNumberService.cls

```diff
@@ -0,0 +1,326 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @description Provides functionality related to custom AutoNumbers for an SObject.
+* Custom AutoNumbers provide the ability to have an "auto number" field that can
+* switch between Display Formats.  Standard Auto Number fields allow changing of the
+* Display Format but the number increments regardless of the format.  Custom AutoNumbers
+* use the active AutoNumber__c record when determining which Display Format to use, and
+* each Display Format keeps track of its Max Last Used number.  This allows it to
+* increment based on the last used number for the active Display Format, rather than
+* using a global number sequence for all new records for the object.
+*/
+public class AutoNumberService {
+    SObjectType sObjType;
+    List<SObject> records;
+    public class AutoNumberException extends Exception {}
+
+    /**
+     * @description This constructor is currently used by the settings UI
+     * to create and activate/deactivate Auto Numbers for the passed in
+     * SObjectType.
+     */
+    public AutoNumberService(SObjectType sObjType) {
+        this.sObjType = sObjType;
+    }
+
+    /**
+     * @description This constructor is currently used by the Trigger Handler
+     * framework when applying Auto Numbers to new records.
+     */
+    public AutoNumberService(List<SObject> records) {
+        this.records = records;
+        this.sObjType = records.isEmpty() ? null : records[0].getSObjectType();
+    }
+
+    public void activate(String autoNumberId) {
+        List<AutoNumber__c> autoNumbersToUpdate = new List<AutoNumber__c>();
+        for (AutoNumber__c autoNumber : activeAutoNumbers) {
+            autoNumber.IsActive__c = false;
+            autoNumbersToUpdate.add(autoNumber);
+        }
+
+        autoNumbersToUpdate.add(new AutoNumber__c(Id = autoNumberId, IsActive__c = true));
```

Comment:
Just to confirm I'm following, this will basically update the existing record by ID without touching the other AutoNumber__c fields? 👍

Reply:
Yep!